tekton/publish-nightly.yaml

```diff
@@ -0,0 +1,226 @@
+apiVersion: tekton.dev/v1alpha1
```

Comment:
There are a few differences between publish and publish-nightly, but not so many, so it would be nice eventually to have them converge to a single Task executed with different inputs.

One difference that might be difficult to overcome is the version of the ko-ci image, which is `gcr.io/tekton-releases/ko-ci` in one case and `gcr.io/tekton-nightly/ko-ci` in the other case. I wonder if we support variable substitution in the `image` field as well?